src/org/rascalmpl/uri/file/MavenRepositoryURIResolver.java

```diff
@@ -134,87 +133,174 @@ private static String computeMavenCommandName() {
      */
     private static String getLocalRepositoryLocationFromMavenCommand() {
         try {
-            ProcessBuilder processBuilder = new ProcessBuilder(computeMavenCommandName(), 
+            var tempFile = Maven.getTempFile("classpath");
+
+            Maven.runCommand(List.of(
                 "-q", 
                 "help:evaluate",
                 "-Dexpression=settings.localRepository",
-                "-DforceStdout"
-               );
-            processBuilder.environment().put("JAVA_HOME", System.getProperty("java.home", System.getenv("JAVA_HOME")));
-
-            Process process = processBuilder.start();
-            if (process.waitFor() != 0) {
-                throw new IOException("could not run mvn to retrieve local repository location");
-            }
+                "-Doutput=" + tempFile
+               ), null /* no current pom.xml file */, tempFile);
             
-            try (var reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+               
+            try (var reader = new BufferedReader(new FileReader(tempFile.toString()))) {
                 return reader.lines().collect(Collectors.joining()).trim(); 
             }
         }
-        catch (IOException | InterruptedException e) {
+        catch (IOException e) {
             // it's ok to fail. that just happens.
             return null;
         }
     }
+ 
+    /** 
+     * @param input   mvn://groupid!artifactId!version/path
+     * @return        a file:/// reference to the jar file that is designated by the authority.
+     * @throws IOException when the authority does not designate a jar file
+     */
+    private ISourceLocation resolveOutside(ISourceLocation input) throws IOException {
+        String authority = input.getAuthority();
+
+        if (authority.isEmpty()) {
+            throw new IOException("missing mvn://groupid!artifactId!version/ as the authority in " + input);
+        }
+
+        var parts = authority.split("!");
```

Comment:
On main it's now `--` so after this PR it flips back to `!`? I thought we decided to migrate to `--`?

Reply:
yep